src/main/java/world/reports/NPopulatedCitiesWithinRegion.java

```diff
@@ -19,14 +19,16 @@
 
             String strSelect = "SELECT ci.Name, ci.Population" +
                     "FROM city ci " +
+                    "JOIN country c on ci.country = c.name" +
+                    "WHERE c.region = REGION" +
                     "ORDER BY ci.Population DESC;" +
                     "LIMIT N";
```

Comment:
Then that user input will need to be added to the SQL statement, which you can do by first writing your SQL so that it accepts parameters.
For the coursework, we can just use the `?` mark to accept parameters in MySQL.
Somehow like this:
```suggestion
                    "LIMIT ?";
```